Engine/SerializerPlugins/IArtifactListener.cs

```diff
@@ -0,0 +1,8 @@
+using System.IO;
+namespace OpenTap.Plugins
+{
+    public interface IArtifactListener{
+        void OnArtifactPublished(TestRun run, Stream s, string filename);
```

Comment:
Do we really need both of these? Could we do with this one? Also, should we rename filename to name?